### PR TITLE
[Storage] `az storage account local-user`: Support nfsv3 and extended-group

### PR DESCRIPTION
**Pull request by calvinhzy**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### Related command

`az storage account create/update`: Support `--enable-extended-groups`
`az storage account local-user create/update`: Support `--extended-group`
`az storage account local-user create`: Support `--is-nfsv3-enabled`

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally? (`pip install wheel==0.30.0` required)
- [ ] My extension version conforms to the [Extension version schema](https://github.com/Azure/azure-cli/blob/release/doc/extensions/versioning_guidelines.md)

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
Thank you for your contribution! We will review the pull request and get back to you soon.

File: src/storage-preview/azext_storage_preview/_params.py [hunk 1]
```diff
@@ -315,6 +315,8 @@ def load_arguments(self, _):  # pylint: disable=too-many-locals, too-many-statem
         c.argument('public_network_access', arg_type=get_enum_type(public_network_access_enum), min_api='2021-06-01',
                    help='Enable or disable public network access to the storage account. '
                         'Possible values include: `Enabled` or `Disabled`.')
+        c.argument('enable_extended_groups', arg_type=get_three_state_flag(),
+                   help='Enables extended group support with local users feature, if set to true.')
```
Comment (evelyn-ys) [inline src/storage-preview/azext_storage_preview/_params.py:319]:
```suggestion
                   help='Enable extended group support with local users feature, if set to true.')
```

File: src/storage-preview/azext_storage_preview/_params.py [hunk 2]
```diff
@@ -387,6 +389,8 @@ def load_arguments(self, _):  # pylint: disable=too-many-locals, too-many-statem
                         'Possible values include: `Enabled` or `Disabled`.')
         c.argument('account_name', acct_name_type, options_list=['--name', '-n'])
         c.argument('resource_group_name', required=False, validator=process_resource_group)
+        c.argument('enable_extended_groups', arg_type=get_three_state_flag(),
+                   help='Enables extended group support with local users feature, if set to true.')
```
Comment (evelyn-ys) [inline src/storage-preview/azext_storage_preview/_params.py:393]:
```suggestion
                   help='Enable extended group support with local users feature, if set to true.')
```

File: src/storage-preview/azext_storage_preview/_params.py [hunk 3]
```diff
@@ -949,3 +953,9 @@ def load_arguments(self, _):  # pylint: disable=too-many-locals, too-many-statem
                        arg_type=get_three_state_flag(),
                        help='Indicates whether ACL authorization is allowed for this user. '
                             'Set it to false to disallow using ACL authorization.')
+            c.argument('extended_groups', nargs='+',
+                       help='Supplementary group membership. Only applicable for local users enabled for NFSv3 access.')
+
+    with self.argument_context('storage account local-user create') as c:
+        c.argument('is_nfsv3_enabled', arg_type=get_three_state_flag(),
+                   help='Indicates if the local user is enabled for access with NFSv3 protocol.')
```
Comment (evelyn-ys) [inline src/storage-preview/azext_storage_preview/_params.py:961]:
```suggestion
                   help='Indicate if the local user is enabled for access with NFSv3 protocol.')
```

File: src/storage-preview/azext_storage_preview/_params.py [hunk 4]
```diff
@@ -949,3 +953,9 @@ def load_arguments(self, _):  # pylint: disable=too-many-locals, too-many-statem
                        arg_type=get_three_state_flag(),
                        help='Indicates whether ACL authorization is allowed for this user. '
```
Comment (evelyn-ys) [inline src/storage-preview/azext_storage_preview/_params.py:954]:
```suggestion
                       help='Indicate whether ACL authorization is allowed for this user. '
```

**Review by evelyn-ys — APPROVED** [review]

**Pull request merged**